### ISSUE TITLE
ol/layer/VectorTile  clip layer based on a vector source, the labels is not be clipped

### ISSUE THREAD

**Issue by songyumeng** [openlayers/openlayers#11654]
**Describe the bug**
ol/layer/VectorTile  clip layer based on a vector source, just like [layer-clipping-vector](https://openlayers.org/en/v6.4.3/examples/layer-clipping-vector.html),but labels is not be clipped.
BTW, In ol/layer/VectorLayer with labels clip is worked

**To Reproduce**
1. Go to [demo](https://codesandbox.io/s/vectort-6v84i?file=/src/index.js)

**Expected behavior**
ol/layer/VectorTile  clip layer based on a vector source, the labels is also clipped

**Comment by ahocevar** [comment]
Because of the way decluttering works, this behavior is expected when the layer is configured with `declutter: true`. But without decluttering, clipping should work as expected.

**Comment by songyumeng** [comment]
> But without decluttering, clipping should work as expected.

I think maybe the reason is fire `postrender` before the text is rendered.
https://github.com/openlayers/openlayers/blob/main/src/ol/renderer/canvas/VectorTileLayer.js#L648

> Because of the way decluttering works, this behavior is expected when the layer is configured with declutter: true

Is there any way to make this behavior worked when `declutter: true`

**Comment by mike-000** [comment]
See #11606 for a workaround.  With the new per-layer decluttering logic `prerenderdeclutter` and `postrenderdeclutter` events could also be added?

**Comment by ahocevar** [comment]
> With the new per-layer decluttering logic `prerenderdeclutter` and `postrenderdeclutter` events could also be added?

Like I said already in https://github.com/openlayers/openlayers/issues/11606#issuecomment-700182699, I think It should be enough to trigger the `postrender` event after `renderDeclutter()`, when there are declutter executor groups.

**Comment by mike-000** [comment]
Yes, as long as className is used where appropriate that would be sufficient.

**Comment by songyumeng** [comment]
hi,
Is there any new progress?
I tried the latest version v6.5.0 and still have this problem.
